### PR TITLE
Gather metrics for UOK test results and treat them as failures

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

File: testsuite/gnat2goto/testsuite.py [hunk 1]
```diff
@@ -52,25 +53,39 @@ def main():
...
-    collect_result = generate_collect_result(
+    # Generate a standard 'collect_result' function...
+    generated_collect_result = generate_collect_result(
         result_dir = m.options.output_dir,
         results_file = m.options.results_file,
         output_diff = m.options.view_diffs,
         metrics = test_metrics)
 
+    # ... and then wrap that generated 'collect_result' function in something
+    # that will also accumualte 'UOK' test results
```
Comment (xbauch) [inline testsuite/gnat2goto/testsuite.py]: accumulate

Comment (chrisr-diffblue) [reply testsuite/gnat2goto/testsuite.py]: :-) Thanks my dodgy typing striking again...

File: testsuite/gnat2goto/testsuite.py [hunk 2]
```diff
@@ -52,25 +53,39 @@ def main():
...
         result_dir = m.options.output_dir,
         results_file = m.options.results_file,
         output_diff = m.options.view_diffs,
         metrics = test_metrics)
 
+    # ... and then wrap that generated 'collect_result' function in something
+    # that will also accumulate 'UOK' test results
+    def collect_result_including_uok(name, process, _job_info):
+        generated_collect_result(name, process, _job_info)
+        test_name = test_name = os.path.basename(name)
```
Comment (xbauch) [inline testsuite/gnat2goto/testsuite.py]: Am I missing something?

**Comment by hannes-steffenhagen-diffblue** [comment]
seems ok to, kind of bugs me that we don't have unit tests for our test scripts though :)

File: testsuite/gnat2goto/testsuite.py (hunk 2 above)
Comment (hannes-steffenhagen-diffblue) [reply testsuite/gnat2goto/testsuite.py]: looks like a typo

Comment (xbauch) [reply testsuite/gnat2goto/testsuite.py]: Or some python magic, you never know.

**Comment by chrisr-diffblue** [comment]
@hannes-steffenhagen-diffblue "Who tests the testers?" :-) In an earlier commit in this PR (now removed) I deliberately marked a passing regression test as "XFAIL" and the changes in this PR did cause travis to flag the job as failing because of the 'UOK' result.

File: testsuite/gnat2goto/testsuite.py (hunk 2 above)
Comment (chrisr-diffblue) [reply testsuite/gnat2goto/testsuite.py]: Argh, yes - typo, and I didn't spot the comments here before merging - my bad. I'll do another PR to fix.

**Comment by chrisr-diffblue** [comment]
PR to fix the *ahem* deliberate typo... https://github.com/diffblue/gnat2goto/pull/87

**Pull request merged**